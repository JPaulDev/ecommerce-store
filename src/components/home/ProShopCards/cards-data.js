import proAi from '../../../../public/images/pro-shop/pro-ai.webp';
import proAudio from '../../../../public/images/pro-shop/pro-audio.webp';
import proGaming from '../../../../public/images/pro-shop/pro-gaming.webp';
import proGraphics from '../../../../public/images/pro-shop/pro-graphics.webp';
import proLogos from '../../../../public/images/pro-shop/pro-logos.webp';
import proSignage from '../../../../public/images/pro-shop/pro-signage.webp';
import proVideo from '../../../../public/images/pro-shop/pro-video.webp';

const PRO_SHOP_CARDS = [
  {
    logo: proLogos,
    image: proGaming,
    logoPosition: '-170px',
    description: `The fastest gaming PCs - plus a huge range of peripherals for
      the ultimate gaming experience.`,
    linkText: 'Visit Gaming Store',
    href: '/',
  },
  {
    logo: proLogos,
    image: proGraphics,
    logoPosition: '-510px',
    description: `Workstations and servers for graphics-intensive workflows such
      as AEC & BIM, CAD & CAE and M & E.`,
    linkText: 'Visit Graphics Store',
    href: '/',
  },
  {
    logo: proLogos,
    image: proAudio,
    logoPosition: '0',
    description: `Computer based music and audio production - plus the best in
      music technology and instruments.`,
    linkText: 'Visit Audio Store',
    href: '/',
  },
  {
    logo: proLogos,
    image: proVideo,
    logoPosition: '-340px',
    description: `Film and broadcast equipment, workstations for professional
      video and media production workflows.`,
    linkText: 'Visit Video Store',
    href: '/',
  },
  {
    image: proSignage,
    heading: 'Digital Signage Solutions',
    description: `Reinvent the way you engage with customers with solutions that
      provide invaluable insights and customer data.`,
    linkText: 'View Signage Solutions',
    href: '/',
  },
  {
    image: proAi,
    heading: 'Deep Learning & AI',
    description: `Desktop, datacentre and cloud solutions for developing and
      deploying deep learning and AI models.`,
    linkText: 'Find Out More',
    href: '/',
  },
];

export default PRO_SHOP_CARDS;
