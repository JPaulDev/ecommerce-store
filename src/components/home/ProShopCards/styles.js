import styled from 'styled-components';

export const Section = styled.section`
  background-color: white;
  border: 1px solid var(--color-grey-12);
  display: grid;
  gap: 6px;
  margin-top: 15px;
  padding: 6px;

  @media screen and (min-width: 780px) {
    grid-template-columns: 1fr 1fr;
  }
`;

export const ProCard = styled.div`
  background-color: var(--color-grey-14);
  background-image: url(${({ image }) => image});
  background-position: right;
  background-repeat: no-repeat;
  min-height: 125px;
  padding-left: 25px;
  padding-top: 25px;

  @media screen and (min-width: 780px) {
    min-height: 140px;
  }

  @media screen and (min-width: 990px) {
    min-height: 175px;
  }
`;

export const ProLogo = styled.div`
  background-image: url(${({ logo }) => logo});
  background-position-x: ${({ logoPosition }) => logoPosition};
  background-repeat: no-repeat;
  height: 18px;
  margin-bottom: 10px;
  width: 170px;
`;

export const Heading = styled.h4`
  color: var(--color-blue-4);
  font-family: 'Uni Sans Book', sans-serif;
  font-size: var(--font-size-26);
  font-weight: var(--font-weight-regular);
  line-height: 0.8;
  margin-bottom: 10px;
`;

export const Text = styled.p`
  color: var(--color-grey-4);
  font-size: var(--font-size-14);
  line-height: 1.5;
  margin-bottom: 10px;
  width: 60%;

  @media screen and (max-width: 989px) {
    display: none;
  }
`;
