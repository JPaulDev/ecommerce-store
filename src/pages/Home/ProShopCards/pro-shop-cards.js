import proLogos from '../../../assets/images/pro-shop/pro-logos.webp';
import proGaming from '../../../assets/images/pro-shop/pro-gaming.webp';
import proGraphics from '../../../assets/images/pro-shop/pro-graphics.webp';
import proAudio from '../../../assets/images/pro-shop/pro-audio.webp';
import proVideo from '../../../assets/images/pro-shop/pro-video.webp';
import proSignage from '../../../assets/images/pro-shop/pro-signage.webp';
import proAi from '../../../assets/images/pro-shop/pro-ai.webp';

const cards = [
  {
    logo: proLogos,
    image: proGaming,
    logoPosition: '-170px',
    description: `The fastest gaming PCs - plus a huge range of peripherals for
      the ultimate gaming experience.`,
    linkText: 'Visit Gaming Store',
    url: '',
  },
  {
    logo: proLogos,
    image: proGraphics,
    logoPosition: '-510px',
    description: `Workstations and servers for graphics-intensive workflows such
      as AEC & BIM, CAD & CAE and M & E.`,
    linkText: 'Visit Graphics Store',
    url: '',
  },
  {
    logo: proLogos,
    image: proAudio,
    logoPosition: '0',
    description: `Computer based music and audio production - plus the best in
      music technology and instruments.`,
    linkText: 'Visit Audio Store',
    url: '',
  },
  {
    logo: proLogos,
    image: proVideo,
    logoPosition: '-340px',
    description: `Film and broadcast equipment, workstations for professional
      video and media production workflows.`,
    linkText: 'Visit Video Store',
    url: '',
  },
  {
    image: proSignage,
    heading: 'Digital Signage Solutions',
    description: `Reinvent the way you engage with customers with solutions that
      provide invaluable insights and customer data.`,
    linkText: 'View Signage Solutions',
    url: '',
  },
  {
    image: proAi,
    heading: 'Deep Learning & AI',
    description: `Desktop, datacentre and cloud solutions for developing and
      deploying deep learning and AI models.`,
    linkText: 'Find Out More',
    url: '',
  },
];

export default cards;
