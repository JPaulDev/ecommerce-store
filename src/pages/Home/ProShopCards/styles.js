import styled from 'styled-components';

export const Section = styled.section`
  background-color: white;
  border: 1px solid var(--color-grey-9);
  display: grid;
  gap: 6px;
  margin-top: 15px;
  padding: 6px;

  @media screen and (min-width: 780px) {
    grid-template-columns: 1fr 1fr;
  }
`;

export const ProCard = styled.div`
  background-color: #f1f1f1;
  background-image: url(${({ image }) => image});
  background-position: right;
  background-repeat: no-repeat;
  min-height: 125px;
  padding-left: 25px;
  padding-top: 25px;

  @media screen and (min-width: 780px) {
    min-height: 140px;
  }

  @media screen and (min-width: 990px) {
    min-height: 175px;
  }
`;

export const ProLogo = styled.div`
  background-image: url(${({ logo }) => logo});
  background-position-x: ${({ logoPosition }) => logoPosition};
  background-repeat: no-repeat;
  height: 18px;
  margin-bottom: 10px;
  width: 170px;
`;

export const Heading = styled.h4`
  color: #0072bc;
  font-family: 'Uni Sans Book', sans-serif;
  font-size: 1.7rem;
  font-weight: 400;
  line-height: 0.8;
  margin-bottom: 10px;
`;

export const Text = styled.p`
  color: var(--color-grey-8);
  font-size: 0.9rem;
  margin-bottom: 10px;
  width: 60%;
`;
