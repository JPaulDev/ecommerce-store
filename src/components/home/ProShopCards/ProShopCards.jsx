import { LinkChevron } from '../../ui';
import cards from './pro-shop-cards';
import * as Styled from './styles';

export default function ProShopCards() {
  return (
    <Styled.Section>
      {cards.map((item) => (
        <Styled.ProCard key={item.linkText} image={item.image.src}>
          {item.logo ? (
            <Styled.ProLogo
              logo={item.logo.src}
              logoPosition={item.logoPosition}
            />
          ) : (
            <Styled.Heading>{item.heading}</Styled.Heading>
          )}
          <Styled.Text>{item.description}</Styled.Text>
          <LinkChevron href={item.href} fontSize={14} iconSize={20}>
            {item.linkText}
          </LinkChevron>
        </Styled.ProCard>
      ))}
    </Styled.Section>
  );
}
