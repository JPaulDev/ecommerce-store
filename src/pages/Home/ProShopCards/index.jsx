import useMediaQuery from '../../../hooks/useMediaQuery';
import LinkWithChevron from '../../../components/LinkWithChevron';
import cards from './pro-shop-cards';
import * as Styled from './styles';

export default function ProShopCards() {
  const isMatch = useMediaQuery('(min-width: 990px)');

  return (
    <Styled.Section>
      {cards.map((item) => (
        <Styled.ProCard key={item.linkText} image={item.image}>
          {item.logo ? (
            <Styled.ProLogo logo={item.logo} logoPosition={item.logoPosition} />
          ) : (
            <Styled.Heading>{item.heading}</Styled.Heading>
          )}
          {isMatch && <Styled.Text>{item.description}</Styled.Text>}
          <LinkWithChevron
            url={item.url}
            text={item.linkText}
            fontSize="0.9rem"
            iconSize="21px"
            marginTop="10px"
          />
        </Styled.ProCard>
      ))}
    </Styled.Section>
  );
}
