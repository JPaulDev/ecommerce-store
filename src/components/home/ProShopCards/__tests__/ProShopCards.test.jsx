import renderer from 'react-test-renderer';
import ProShopCards from '../index';

describe('ProShopCards component', () => {
  it('renders 6 pro shop cards with links and shop descriptions', () => {
    const tree = renderer.create(<ProShopCards />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
