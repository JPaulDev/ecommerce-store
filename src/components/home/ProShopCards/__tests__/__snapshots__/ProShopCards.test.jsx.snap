// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProShopCards component renders 6 pro shop cards with links and shop descriptions 1`] = `
.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: var(--color-grey-8);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.9rem;
  font-weight: 700;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
}

.c4 svg {
  margin-left: 10px;
}

.c0 {
  background-color: white;
  border: 1px solid var(--color-grey-9);
  display: grid;
  gap: 6px;
  margin-top: 15px;
  padding: 6px;
}

.c1 {
  background-color: #f1f1f1;
  background-image: url(/img.jpg);
  background-position: right;
  background-repeat: no-repeat;
  min-height: 125px;
  padding-left: 25px;
  padding-top: 25px;
}

.c2 {
  background-image: url(/img.jpg);
  background-position-x: -170px;
  background-repeat: no-repeat;
  height: 18px;
  margin-bottom: 10px;
  width: 170px;
}

.c5 {
  background-image: url(/img.jpg);
  background-position-x: -510px;
  background-repeat: no-repeat;
  height: 18px;
  margin-bottom: 10px;
  width: 170px;
}

.c6 {
  background-image: url(/img.jpg);
  background-position-x: 0;
  background-repeat: no-repeat;
  height: 18px;
  margin-bottom: 10px;
  width: 170px;
}

.c7 {
  background-image: url(/img.jpg);
  background-position-x: -340px;
  background-repeat: no-repeat;
  height: 18px;
  margin-bottom: 10px;
  width: 170px;
}

.c8 {
  color: #0072bc;
  font-family: 'Uni Sans Book',sans-serif;
  font-size: 1.7rem;
  font-weight: 400;
  line-height: 0.8;
  margin-bottom: 10px;
}

.c3 {
  color: var(--color-grey-8);
  font-size: 0.9rem;
  margin-bottom: 10px;
  width: 60%;
}

@media screen and (min-width:780px) {
  .c0 {
    grid-template-columns: 1fr 1fr;
  }
}

@media screen and (min-width:780px) {
  .c1 {
    min-height: 140px;
  }
}

@media screen and (min-width:990px) {
  .c1 {
    min-height: 175px;
  }
}

@media screen and (max-width:989px) {
  .c3 {
    display: none;
  }
}

<section
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    />
    <p
      className="c3"
    >
      The fastest gaming PCs - plus a huge range of peripherals for
      the ultimate gaming experience.
    </p>
    <a
      className="c4"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Visit Gaming Store
      <svg
        fill="currentColor"
        height={20}
        viewBox="0 0 512 512"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 256C0 397.4 114.6 512 256 512s256-114.6 256-256S397.4 0 256 0S0 114.6 0 256zM241 377c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l87-87-87-87c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L345 239c9.4 9.4 9.4 24.6 0 33.9L241 377z"
        />
      </svg>
    </a>
  </div>
  <div
    className="c1"
  >
    <div
      className="c5"
    />
    <p
      className="c3"
    >
      Workstations and servers for graphics-intensive workflows such
      as AEC & BIM, CAD & CAE and M & E.
    </p>
    <a
      className="c4"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Visit Graphics Store
      <svg
        fill="currentColor"
        height={20}
        viewBox="0 0 512 512"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 256C0 397.4 114.6 512 256 512s256-114.6 256-256S397.4 0 256 0S0 114.6 0 256zM241 377c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l87-87-87-87c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L345 239c9.4 9.4 9.4 24.6 0 33.9L241 377z"
        />
      </svg>
    </a>
  </div>
  <div
    className="c1"
  >
    <div
      className="c6"
    />
    <p
      className="c3"
    >
      Computer based music and audio production - plus the best in
      music technology and instruments.
    </p>
    <a
      className="c4"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Visit Audio Store
      <svg
        fill="currentColor"
        height={20}
        viewBox="0 0 512 512"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 256C0 397.4 114.6 512 256 512s256-114.6 256-256S397.4 0 256 0S0 114.6 0 256zM241 377c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l87-87-87-87c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L345 239c9.4 9.4 9.4 24.6 0 33.9L241 377z"
        />
      </svg>
    </a>
  </div>
  <div
    className="c1"
  >
    <div
      className="c7"
    />
    <p
      className="c3"
    >
      Film and broadcast equipment, workstations for professional
      video and media production workflows.
    </p>
    <a
      className="c4"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Visit Video Store
      <svg
        fill="currentColor"
        height={20}
        viewBox="0 0 512 512"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 256C0 397.4 114.6 512 256 512s256-114.6 256-256S397.4 0 256 0S0 114.6 0 256zM241 377c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l87-87-87-87c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L345 239c9.4 9.4 9.4 24.6 0 33.9L241 377z"
        />
      </svg>
    </a>
  </div>
  <div
    className="c1"
  >
    <h4
      className="c8"
    >
      Digital Signage Solutions
    </h4>
    <p
      className="c3"
    >
      Reinvent the way you engage with customers with solutions that
      provide invaluable insights and customer data.
    </p>
    <a
      className="c4"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      View Signage Solutions
      <svg
        fill="currentColor"
        height={20}
        viewBox="0 0 512 512"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 256C0 397.4 114.6 512 256 512s256-114.6 256-256S397.4 0 256 0S0 114.6 0 256zM241 377c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l87-87-87-87c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L345 239c9.4 9.4 9.4 24.6 0 33.9L241 377z"
        />
      </svg>
    </a>
  </div>
  <div
    className="c1"
  >
    <h4
      className="c8"
    >
      Deep Learning & AI
    </h4>
    <p
      className="c3"
    >
      Desktop, datacentre and cloud solutions for developing and
      deploying deep learning and AI models.
    </p>
    <a
      className="c4"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Find Out More
      <svg
        fill="currentColor"
        height={20}
        viewBox="0 0 512 512"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 256C0 397.4 114.6 512 256 512s256-114.6 256-256S397.4 0 256 0S0 114.6 0 256zM241 377c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l87-87-87-87c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0L345 239c9.4 9.4 9.4 24.6 0 33.9L241 377z"
        />
      </svg>
    </a>
  </div>
</section>
`;
